### DISCUSSION TITLE
Incorrect dismiss behavior after presenting UIAlertController UIKit

### DISCUSSION THREAD

**Discussion by nmalevich** [pointfreeco/swift-composable-architecture#3434] [General]
### Description

I use AlertState, @Presents and present method of UIViewController for showing alerts
I don't have any problems with first show, but after first dismiss property in my state for alerts isn't nil, it is still previous value. So when I try to show next time, it will be nothing, because state isn't change

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

State of alert after dismissing is nil

### Actual behavior

State of alert after dismissing isn't nil

### Reproducing project

[UIKitAlertDismissButtonBugExample.zip](https://github.com/user-attachments/files/17307777/UIKitAlertDismissButtonBugExample.zip)

### The Composable Architecture version information

1.15.0

### Destination operating system

iOS 17.6.1

### Xcode version information

15.4

### Swift Compiler version information

```shell
swift-driver version: 1.90.11.1 Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
Target: x86_64-apple-macosx14.0
```

**Comment by mbrandonw** [comment]
Hi @nmalevich, you need to make sure to use the `ifLet` operator to integrate the alert's behavior into your feature:

```swift
Reduce { state, action in
  …
}
.ifLet(\.$alert, action: \.alert)
```

That should fix it.

Since this isn't an issue with the library I am going to convert it to a discussion. Please feel free to continue the conversation over there!
